### ISSUE TITLE
[Improvement] More debug info on failed/timed out OpenAI requests

### ISSUE THREAD

**Issue by dsent** [umputun/tg-spam#182]
Some messages are not getting OpenAI score at all, and there's nothing in the debug logs for these.
So it's not clear if the request timed out, or was rejected, but the info doesn't show openai results at all, e.g.:
```
2024/11/27 22:33:01,stdout,"2024/11/27 22:33:01.599 [DEBUG] {bot/spam.go:107 bot.(*SpamFilter).OnMessage} user Эржан Кочербайев is not a spammer, {name: stopword, spam: false, details: not found}, {name: images, spam: false, details: no images without text}, {name: cas, spam: false, details: record not found}, {name: similarity, spam: false, details: 0.00/0.50}, {name: classifier, spam: false, details: probability of ham: 61.48%}
```
If OpenAI returns nonsensical result, that is logged. But in these cases, there's nothing.

**Comment by umputun** [comment]
I don't see how this even possible unless you are using openai check with veto mode

https://github.com/umputun/tg-spam/blob/64c97cc9762d116d47a867481e0c7dc40f662b5d/lib/tgspam/detector.go#L179-L179

this code should add openai spam result regardless of the timeout, rejects and any of this, see the actual check https://github.com/umputun/tg-spam/blob/64c97cc9762d116d47a867481e0c7dc40f662b5d/lib/tgspam/openai.go#L71-L71

So, logically I don't see how this case is even possible unless there is some bug somewhere in those two functions. Any details you can share what will be helpful for me to reproduce it?

**Comment by dsent** [comment]
I actually use OpenAI check with veto mode. I will study the logs around those messages more thoroughly.

Basically, what I'm looking for are the same messages before they were marked manually by admins, when they were originally marked as ham, right?

In this case, when they're marked manually, there's no OpenAI request happening at all, but the other checks run and their results posted as "original test results", is that correct?

If that's the case, the wording should probably be changed to something like "current test results" or "reference re-test results".

**Comment by dsent** [comment]
I'd really like to have a message in admin group when OpenAI vetoes some message from being marked spam (ideally with an options to veto its vetoing 😅).

Right now it's kinda not easy to debug when OpenAI misbehaves.

**Comment by umputun** [comment]
I am uncertain why you expect OpenAI to be involved in veto mode in this scenario. This mode is designed for AI to verify the conclusion that “this was spam for real.” If none of the other checks indicate spam, the OpenAI check will not make a decision to change its status and will not be called upon in this instance.

The sole purpose of veto mode is to confirm (or reject) the spam detection result from other checks. To reiterate, if all other checks assert “ham,” the OpenAI check will not be even invoked.

**Comment by umputun** [comment]
>I'd really like to have a message in admin group when OpenAI vetoes some message from being marked spam (ideally with an option to veto its vetoing 😅).

Makes sense. I added it to master as a log message. This is what it looks like in the test:

`openai vetoed ham message: "some message 1234", checks: [{stopword: spam, some message}, {openai: ham, good text, confidence: 100%}] `

**Comment by dsent** [comment]
Ok, I might have got the wrong idea about veto mode. So, basically, OpenAI is either helps detect spam more aggressively (in normal mode), or helps detect false positives (in veto mode).

I thought that veto mode works **in addition** to a normal check, basically, making OpenAI the main source of classification, with the rest of the checks working as a fallback in case OpenAI fails (times out, returns a wrong result etc.).

**Comment by paskal** [comment]
@dsent could you maybe improve the documentation to make things cleaner for everyone?

https://github.com/umputun/tg-spam/blob/8efa4d61fb852b601d1abf79dc932e4023b22bd3/site/docs/index.md?plain=1#L101C201-L101C205

**Comment by dsent** [comment]
I haven't figured out how to make a pull request without forking the project and creating my own branches there (that's the way to go, probably, but seems a bit bothersome to me), so here is my proposed phrasing:
```
- Only the first message(s) from a given user is checked for spam. If `--paranoid` mode is enabled, openai will not be used at all.
- OpenAI check is the last in the chain of checks. By default (if `--openai.veto` is not set), the bot will not even call OpenAI if any of the previous checks marked the message as spam. This default mode makes spam detection stricter, helping detect more spam messages that otherwise could have slipped through the cracks.
- Setting `--openai.veto` changes the workflow. In veto mode, OpenAI is called *only* if the message is classified as spam by other checks. The message is considered spam only if OpenAI confirms the decision. This helps reduce the number of false positives, making spam detection more careful.
- By default, OpenAI integration is disabled. 
```

**Comment by dsent** [comment]
Just noticed that the text in README.md is now out of sync with site/docs/index.md.

**Comment by paskal** [comment]
Thanks! Fixed in #189.
